infrastructure/restapi/src/test/java/tech/pegasys/teku/infrastructure/restapi/types/EnumTypeDefinitionTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.infrastructure.restapi.types;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.infrastructure.restapi.json.JsonUtil;
+
+public class EnumTypeDefinitionTest {
+  SerializableTypeDefinition<SimpleEnum> definition =
+      new SerializableObjectTypeDefinitionBuilder<SimpleEnum>()
+          .withField(
+              "ANSWER", DeserializableTypeDefinition.enumOf(YesNo.class), SimpleEnum::getYesno)
+          .build();
```

Comment:
I think you actually just want this to be 
```suggestion
  DeserializableTypeDefinition<YesNo> definition = DeserializableTypeDefinition.enumOf(YesNo.class);
```

because you only want to test the enum deserialiser, not the object deserialisation wrapper of it as well.

And the test is probably just that it can round trip each value of the enum. `CoreTypesTest` is probably a useful reference.